lib/commands/cache.js

```diff
@@ -168,7 +168,7 @@ class Cache extends BaseCommand {
       '    npm cache add <folder>...\n'
     log.silly('cache add', 'args', args)
     if (args.length === 0)
-      throw Object.assign(new Error(usage), { code: 'EUSAGE' })
+      throw this.usageError(usage)
```

Comment:
```suggestion
      throw this.usageError()
```

Reply:
This one changes the behavior of the usage error for the add command:

Here's what we do currently:
```sh
❯ npm cache add
npm ERR! code EUSAGE
npm ERR! Usage:
npm ERR!     npm cache add <tarball-url>...
npm ERR!     npm cache add <pkg>@<ver>...
npm ERR!     npm cache add <tarball>...
npm ERR!     npm cache add <folder>...
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lukekarrys/.npm/_logs/2021-11-04T15_37_18_488Z-debug.log
```

With this change:
```sh
❯ node . cache add
npm ERR! code EUSAGE
npm ERR! 
npm ERR! Usage: npm cache
npm ERR! 
npm ERR! Manipulates packages cache
npm ERR! 
npm ERR! Usage:
npm ERR! npm cache add <tarball file>
npm ERR! npm cache add <folder>
npm ERR! npm cache add <tarball url>
npm ERR! npm cache add <git url>
npm ERR! npm cache add <name>@<version>
npm ERR! npm cache clean [<key>]
npm ERR! npm cache ls [<name>@<version>]
npm ERR! npm cache verify
npm ERR! 
npm ERR! Options:
npm ERR! [--cache <cache>]
npm ERR! 
npm ERR! Run "npm help cache" for more info

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lukekarrys/.npm/_logs/2021-11-04T15_37_39_334Z-debug.log
```

But I'm not sure if we should show both messages combined?

```sh
npm ERR! code EUSAGE
npm ERR! 
npm ERR! Usage: Usage:
npm ERR!     npm cache add <tarball-url>...
npm ERR!     npm cache add <pkg>@<ver>...
npm ERR!     npm cache add <tarball>...
npm ERR!     npm cache add <folder>...
npm ERR! 
npm ERR! 
npm ERR! npm cache
npm ERR! 
npm ERR! Manipulates packages cache
npm ERR! 
npm ERR! Usage:
npm ERR! npm cache add <tarball file>
npm ERR! npm cache add <folder>
npm ERR! npm cache add <tarball url>
npm ERR! npm cache add <git url>
npm ERR! npm cache add <name>@<version>
npm ERR! npm cache clean [<key>]
npm ERR! npm cache ls [<name>@<version>]
npm ERR! npm cache verify
npm ERR! 
npm ERR! Options:
npm ERR! [--cache <cache>]
npm ERR! 
npm ERR! Run "npm help cache" for more info

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lukekarrys/.npm/_logs/2021-11-04T15_38_30_550Z-debug.log
```

Reply:
Ah yeah that one's a bit ugly cause we don't have proper subcommands.  We can leave this one, I didn't catch that it was building a custom message there.